src/Compilers/Core/Portable/Symbols/Attributes/CommonAttributeData.cs

```diff
@@ -196,8 +194,23 @@ internal ConstantValue DecodeDateTimeConstantValue()
 
         #endregion
 
+        internal ObsoleteAttributeData DecodeObsoleteAttribute(ObsoleteAttributeKind kind)
+        {
+            switch (kind)
+            {
+                case ObsoleteAttributeKind.Obsolete:
+                    return DecodeObsoleteAttribute();
+                case ObsoleteAttributeKind.Deprecated:
+                    return DecodeDeprecatedAttribute();
+                case ObsoleteAttributeKind.Experimental:
+                    return DecodeExperimentalAttribute();
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(kind);
+            }
+        }
+
         /// <summary>
-        /// Decode the arguments to ObsoleteAttribute. ObsoleteAttribute can have 0,1 or 2 arguments.
+        /// Decode the arguments to ObsoleteAttribute. ObsoleteAttribute can have 0, 1 or 2 arguments.
         /// </summary>
         internal ObsoleteAttributeData DecodeObsoleteAttribute()
```

Comment:
Are there still references to this method, or do we funnel all callers through the new overload above (line 197)?